pygaggle/run/evaluate_passage_reader.py

```diff
@@ -124,29 +126,33 @@ def main():
 
     evaluator = ReaderEvaluator(reader)
 
+    max_topk_passages = max(options.topk_em)
     examples = []
     for _, item in data.items():
         examples.append(
             RetrievalExample(
                 query=Query(text=item["question"]),
                 texts=list(map(lambda context: Text(text=context["text"].split('\n', 1)[1].replace('""', '"'),
                                                     title=context["text"].split('\n', 1)[0].replace('"', '')),
-                               item["contexts"]))[:options.use_top_k_passages],
+                               item["contexts"]))[: max_topk_passages],
                 ground_truth_answers=item["answers"],
             )
         )
-    dpr_predictions = [] if args.output_file is not None else None
+    dpr_predictions = { k : [] for k in options.topk_em } if args.output_dir is not None else None
 
     ems = evaluator.evaluate(examples, dpr_predictions)
 
     logging.info(f'Reader completed')
 
-    em = np.mean(np.array(ems)) * 100.
-    logging.info(f'Exact Match Accuracy: {em}')
+    for k in options.topk_em:
+        em = np.mean(np.array(ems[k])) * 100.
+        logging.info(f'Top{k}\tExact Match Accuracy: {em}')
 
-    if args.output_file is not None:
-        with open(args.output_file, 'w', encoding='utf-8', newline='\n') as f:
-            json.dump(dpr_predictions, f, indent=4)
+    if args.output_dir is not None:
+        os.makedirs(args.output_dir, exist_ok=True)
+        for k in options.topk_em:
+            with open(os.path.join(args.output_dir, f'dpr_predictions_top{k}'), 'w', encoding='utf-8', newline='\n') as f:
+                json.dump(dpr_predictions[k], f, indent=4)
```

Comment:
no need to generate multiple files.
I would prefer store in one json. 
each item in following format.
{
"question": <>, "answers": {"top10": <>, "top20": <> ...}
}